File: dodam-system-domain/dodam-domain-rds/src/main/java/b1nd/dodam/domain/rds/banner/repository/BannerRepository.java
package b1nd.dodam.domain.rds.banner.repository;

import b1nd.dodam.domain.rds.banner.entity.Banner;
import b1nd.dodam.domain.rds.banner.enumeration.BannerStatus;
import b1nd.dodam.domain.rds.banner.exception.BannerNotFoundException;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.List;

public interface BannerRepository extends JpaRepository<Banner, Integer> {

    List<Banner> findAllByOrderByIdDesc();

    List<Banner> findByStatusOrderByIdDesc(BannerStatus bannerStatus);

    default Banner getById(int id) {
        return findById(id)
                .orElseThrow(BannerNotFoundException::new);
    }

}

File: services/service-banner/src/main/kotlin/com/b1nd/dodamdodam/banner/domain/banner/repository/BannerRepository.kt
package com.b1nd.dodamdodam.banner.domain.banner.repository

import com.b1nd.dodamdodam.banner.domain.banner.entity.BannerEntity
import com.b1nd.dodamdodam.banner.domain.banner.enumeration.BannerStatus
import org.springframework.data.jpa.repository.JpaRepository

interface BannerRepository : JpaRepository<BannerEntity, Long> {
    fun findAllByOrderByIdDesc(): List<BannerEntity>
    fun findByStatusOrderByIdDesc(status: BannerStatus): List<BannerEntity>
}
